### PR TITLE
cleaning(simple-kernel-timer): promote `find_index` to `kp_shared.hpp`

### PR DESCRIPTION
**Pull request by romintomasetti**
As suggested by @masterleinad in #225.

### CODE REVIEW

**Comment by romintomasetti** [comment]
@masterleinad I have `clang-format-14`, not `8`... So I don't know what changes are required... Do you know how we could make the action show a suggested *diff* in the action log? Thanks!

File: profiling/simple-kernel-timer/kp_shared.h [hunk 1]
```diff
@@ -39,6 +42,15 @@ inline bool compareKernelPerformanceInfo(KernelPerformanceInfo* left,
   return left->getTime() > right->getTime();
 };
 
+inline int find_index(std::vector<KernelPerformanceInfo*>& kernels,
```
Comment (dalg24) [inline profiling/simple-kernel-timer/kp_shared.h]: take the vector by const ref

File: profiling/simple-kernel-timer/kp_kernel_info.h [hunk 2]
```diff
@@ -218,13 +214,13 @@ class KernelPerformanceInfo {
...
   char* kernelName;
   // const char* regionName;
-  uint64_t callCount;
-  double time;
-  double timeSq;
-  double startTime;
+  uint64_t callCount = 0;
+  double time        = 0;
+  double timeSq      = 0;
+  double startTime   = 0;
```
Comment (dalg24) [inline profiling/simple-kernel-timer/kp_kernel_info.h]: How are these changes related to sharing the implementation of `find_index`?

**Comment by masterleinad** [comment]
> @masterleinad I have `clang-format-14`, not `8`... So I don't know what changes are required... Do you know how we could make the action show a suggested _diff_ in the action log? Thanks!

See #228.

File: profiling/simple-kernel-timer/kp_shared.h (hunk 1 above)
Comment (romintomasetti) [reply profiling/simple-kernel-timer/kp_shared.h]: OK, done :heavy_check_mark:

File: profiling/simple-kernel-timer/kp_kernel_info.h (hunk 2 above)
Comment (romintomasetti) [reply profiling/simple-kernel-timer/kp_kernel_info.h]: Indeed, reverted :+1:

**Review by masterleinad — APPROVED** [review]

**Comment by romintomasetti** [comment]
@dalg24 Could you have a look and merge if all is fine for you? Thanks !

File: profiling/simple-kernel-timer/kp_shared.h [hunk 3]
```diff
@@ -39,6 +42,15 @@ inline bool compareKernelPerformanceInfo(KernelPerformanceInfo* left,
   return left->getTime() > right->getTime();
 };
 
+inline int find_index(const std::vector<KernelPerformanceInfo*>& kernels,
+                      const char* kernelName) {
+  const auto it =
+      std::find_if(kernels.begin(), kernels.end(), [&](const auto& kernel) {
+        return strcmp(kernel->getName(), kernelName) == 0;
+      });
+  return it == kernels.cend() ? -1 : std::distance(kernels.begin(), it);
```
Comment (dalg24) [inline profiling/simple-kernel-timer/kp_shared.h]: I see little benefit from changing the original implementation
```suggestion
  for (unsigned int i = 0; i < kernels.size(); i++) {
    KernelPerformanceInfo* nextKernel = kernels[i];

    if (strcmp(nextKernel->getName(), kernelName) == 0) {
      return i;
    }
  }

  return -1;
```

File: profiling/simple-kernel-timer/kp_shared.h [hunk 4]
```diff
@@ -17,8 +17,11 @@
 #ifndef _H_KOKKOSP_KERNEL_SHARED
 #define _H_KOKKOSP_KERNEL_SHARED
 
+#include <algorithm>
```
Comment (dalg24) [inline profiling/simple-kernel-timer/kp_shared.h]:
```suggestion
```

File: profiling/simple-kernel-timer/kp_shared.h (hunk 3 above)
Comment (dalg24) [reply profiling/simple-kernel-timer/kp_shared.h]: I meant to apply the suggestion and merge but somehow I don't see the option to do so in the GH interface.

**Review by dalg24 — APPROVED** [review]
